dds/DCPS/RTPS/Sedp.h

```diff
@@ -550,38 +632,49 @@ class Sedp : public DCPS::EndpointManager<ParticipantData_t> {
     virtual ~TypeLookupReader();
   };
 
-  typedef DCPS::RcHandle<TypeLookupReader> TypeLookupReader_rch;
 
 
-  class TypeLookupRequestReader : public TypeLookupReader {
+  class TypeLookupRequestReader : public Reader {
   public:
     TypeLookupRequestReader(const DCPS::RepoId& sub_id, Sedp& sedp)
-      : TypeLookupReader(sub_id, sedp)
+      : Reader(sub_id, sedp)
     {}
 
     virtual ~TypeLookupRequestReader();
 
-    DDS::ReturnCode_t take_tl_request(const DCPS::ReceivedDataSample& sample,
+  private:
+    virtual void data_received_i(const DCPS::ReceivedDataSample& sample,
+      const DCPS::EntityId_t& entity_id,
       DCPS::Serializer& ser,
-      XTypes::TypeLookup_Request& type_lookup_request);
+      const DCPS::Extensibility extensibility);
+
+    DDS::ReturnCode_t process_tl_request(DCPS::Serializer& ser,
+      XTypes::TypeLookup_Reply& type_lookup_reply);
+
+    DDS::ReturnCode_t process_get_types_request(const XTypes::TypeLookup_Request& type_lookup_request,
+      XTypes::TypeLookup_Reply& type_lookup_reply);
   };
 
   typedef DCPS::RcHandle<TypeLookupRequestReader> TypeLookupRequestReader_rch;
 
 
   TypeLookupRequestReader_rch type_lookup_request_reader_;
 
-  class TypeLookupReplyReader : public TypeLookupReader {
+  class TypeLookupReplyReader : public Reader {
   public:
     TypeLookupReplyReader(const DCPS::RepoId& sub_id, Sedp& sedp)
-      : TypeLookupReader(sub_id, sedp)
+      : Reader(sub_id, sedp)
     {}
 
     virtual ~TypeLookupReplyReader();
 
-    DDS::ReturnCode_t take_tl_reply(const DCPS::ReceivedDataSample& sample,
+  private:
+    virtual void data_received_i(const DCPS::ReceivedDataSample& sample,
+      const DCPS::EntityId_t& entity_id,
       DCPS::Serializer& ser,
-      XTypes::TypeLookup_Reply& type_lookup_reply);
+      const DCPS::Extensibility extensibility);
```

Comment:
```suggestion
      DCPS::Extensibility extensibility);
```
multiple occurrences of this one